### PR TITLE
#4501, Identify MySQL syntax allowed in Prepared Statements

### LINKED ISSUE #4501: Identify MySQL syntax allowed in Prepared Statements
### Which version of ShardingSphere did you use?
5.0.0-RC1
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
Sharding-Proxy
### Expected behavior
Use MySQL C API to prepare and execute the SQL 'BEGIN' right.
### Actual behavior
C API return an unknown exception.
### Reason analyze (If you can)
C API expect a ER_UNSUPPORTED_PS message. But Proxy return a success message for now.

### PR DESCRIPTION
**Pull request by tuohai666**
Fixes #4501.

### CODE REVIEW

File: sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareExecutor.java [hunk 1]
```diff
@@ -17,38 +17,79 @@
...
+        SQL_SYNTAX_ALLOWED.addAll(Arrays.asList("ALTER" + "TABLE", "ALTER" + "USER", "ANALYZE" + "TABLE",
+            "CACHE" + "INDEX", "CALL", "CHANGE" + "MASTER", "CHECKSUM" + "TABLE", "CHECKSUM" + "TABLES",
+            "COMMIT", "CREATE" + "INDEX", "DROP" + "INDEX", "CREATE" + "DATABASE", "RENAME" + "DATABASE",
+            "DROP" + "DATABASE", "CREATE" + "TABLE", "DROP" + "TABLE", "CREATE" + "USER", "RENAME" + "USER",
+            "DROP" + "USER", "CREATE" + "VIEW", "DROP" + "VIEW", "DELETE", "DO", "FLUSH" + "TABLE",
+            "FLUSH" + "TABLES", "FLUSH" + "TABLES" + "WITH" + "READ" + "LOCK", "FLUSH" + "HOSTS",
+            "FLUSH" + "PRIVILEGES", "FLUSH" + "LOGS", "FLUSH" + "STATUS", "FLUSH" + "MASTER", "FLUSH" + "SLAVE",
+            "FLUSH" + "DES_KEY_FILE", "FLUSH" + "USER" + "RESOURCES", "GRANT", "INSERT", "INSTALL" + "PLUGIN",
+            "KILL", "LOAD" + "INDEX" + "INTO" + "CACHE", "OPTIMIZE" + "TABLE", "RENAME" + "TABLE",
+            "REPAIR" + "TABLE", "REPLACE", "RESET" + "MASTER", "RESET" + "SLAVE", "RESET" + "QUERY CACHE",
```
Comment (terrymanu) [inline sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareExecutor.java]: Can we move the SQL parser logic into parser module?

Comment (tuohai666) [reply sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareExecutor.java]: Yes, the logic should be moved to parser module.

Comment (tuohai666) [reply sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareExecutor.java]: Finish.

**Pull request merged**